File: JsCode/leetcode/easy/IntersectionOfTwoArrays.js
/**
 * 349. Intersection of Two Arrays
 * https://leetcode.com/problems/intersection-of-two-arrays/description/
 *
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[]}
 *
 * 数组相关方法：
 * indexOf() 搜索数组中的元素，并返回它所在的位置，如果没有该元素则返回-1
 * push() 向数组的末尾添加一个或更多元素，并返回新的长度。
 */
var intersection = function(nums1, nums2) {
    var temp1 = [];
    for(var i = 0; i < nums1.length; i++){
        if (temp1.indexOf(nums1[i]) === -1) {
            temp1.push(nums1[i]);
        }
    }
    var temp2 = [];
    for(var j = 0; j < nums2.length; j++){
        if (temp2.indexOf(nums2[j]) === -1) {
            temp2.push(nums2[j]);
        }
    }
    var result = [];
    for (var k = 0; k < temp2.length; k++){
        if (temp1.indexOf(temp2[k]) !== -1) {
            result.push(temp2[k])
        }
    }
    return result;
};

var arrA = [1, 2, 2, 1];
var arrB = [2, 2];
var result = intersection(arrA, arrB);
for(var t in result) {
    console.log(result[t]);
}

File: PythonCode/leetcode/easy/IntersectionOfTwoArrays.py
#!/usr/bin/python3
# -*- coding: utf-8 -*-

# @Time    : 2018/2/6 15:22
# @Author  : bearyang
# @Site    : 
# @File    : IntersectionOfTwoArrays.py
# @Software: PyCharm
# 349. Intersection of Two Arrays
# https://leetcode.com/problems/intersection-of-two-arrays/description/


class Solution:
    def intersection(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """

        # 先去掉nums1和nums2中的重复元素，存为temp1和temp2
        temp1 = []
        len1 = len(nums1)
        for i in range(len1):
            if nums1[i] not in temp1:
                temp1.append(nums1[i])

        temp2 = []
        len2 = len(nums2)
        for i in range(len2):
            if nums2[i] not in temp2:
                temp2.append(nums2[i])

        result = []
        len2 = len(temp2)
        for i in range(len2):
            if temp2[i] in temp1:
                result.append(temp2[i])
        return result


test = Solution()
arrA = [1, 2, 2, 1]
arrB = [2, 2]
result = test.intersection(arrA, arrB)
for item in result:
    print(item)



